c/registerProduct.c

```diff
@@ -0,0 +1,200 @@
+#include "zosDiscovery.h"
+#endif
+#include "charsets.h"
+#include "plugins.h"
+#ifdef __ZOWE_OS_ZOS
+#include "datasetjson.h"
+#include "authService.h"
+#include "securityService.h"
+#include "zis/client.h"
+#endif
+
+#include "zssLogging.h"
+#include "serviceUtils.h"
+#include "unixFileService.h"
+#include "omvsService.h"
+#include "datasetService.h"
+#include "serverStatusService.h"
+#include "rasService.h"
+#include "certificateService.h"
+#include "registerProduct.h"
+
+#include "jwt.h"
+
+#define PRODUCT "ZLUX"
+#ifndef PRODUCT_MAJOR_VERSION
+#define PRODUCT_MAJOR_VERSION 0
+#endif
+#ifndef PRODUCT_MINOR_VERSION
+#define PRODUCT_MINOR_VERSION 0
+#endif
+#ifndef PRODUCT_REVISION
+#define PRODUCT_REVISION 0
+#endif
+#ifndef PRODUCT_VERSION_DATE_STAMP
+#define PRODUCT_VERSION_DATE_STAMP 0
+#endif
+
+unsigned long long __registerProduct(const char *major_version,
+                                const char *pid,
+                                const char *product_owner,
+                                const char *product_name,
+                                const char *feature_name){
+
+  /* Creates buffers for registration product info */
+
+  char str_product_owner[17];
+  char str_feature_name[17];
+  char str_product_name[17];
+  char str_pid[9];
+  char version[9];
+  unsigned long long ifausage_rc = 0;
+  IFAARGS_t *arg;
+
+  /* Check if SMF is Active first  */
+  char *xx = ((char *__ptr32 *__ptr32 *)0)[4][49];
+  if (0 == xx) {
+    return 1;
+  }
+  if (0 == (*xx & 0x04)) {
+    return 2;
+  }
+
+  /* Left justify with space padding */
+
+  snprintf(str_product_owner, sizeof(str_product_owner), "%-16s",
+           product_owner);
+  snprintf(str_feature_name, sizeof(str_feature_name), "%-16s", feature_name);
+  snprintf(str_product_name, sizeof(str_product_name), "%-16s", product_name);
+  snprintf(str_pid, sizeof(str_pid), "%-8s", pid);
+  snprintf(version, sizeof(version), "%-8s", major_version);
+  /* Register Product with IFAUSAGE */
+
+/*IFAARGS_t *arg = (IFAARGS_t *)__malloc31(sizeof(IFAARGS_t)); */
+  arg = (IFAARGS_t *)__malloc31(sizeof(IFAARGS_t));
+
+  assert(arg);
+  memset(arg, 0, sizeof(IFAARGS_t));
+  memcpy(arg->id, MODULE_REGISTER_USAGE, 8);
+  arg->listlen = sizeof(IFAARGS_t);
+  arg->version = 1;
+  arg->request = 1; /* 1=REGISTER */
+
+  /* Insert properties */
+
+  memcpy(arg->prodowner, str_product_owner, 16);
+  memcpy(arg->prodname, str_product_name, 16);
+  memcpy(arg->prodvers, version, 8);
+  memcpy(arg->prodqual, "NONE    ", 8);
+  memcpy(arg->prodid, str_pid, 8);
+  arg->domain = 1;
+  arg->scope = 1;
+
+
+  arg->prtoken_addr = (char *__ptr32)__malloc31(sizeof(char *__ptr32));
+  arg->begtime_addr = (char *__ptr32)__malloc31(sizeof(char *__ptr32));
+  /* Load 25 (IFAUSAGE) into reg15 and call via SVC */
+  asm(" svc 109\n" : "=NR:r15"(ifausage_rc) : "NR:r1"(arg), "NR:r15"(25) :);
+  free(arg);
+
+  return ifausage_rc;
+
+}
+
+void registerProduct(char *productReg, char *productPID, char *productVer,
```

Comment:
This needs to be declared in the header. Please also provide a description of this function, e.g. what it does, what it returns, etc. You can see examples in [zowe-common-c/h/radmin.h](https://github.com/zowe/zowe-common-c/blob/master/h/radmin.h#L302)

Reply:
yes, will do.

Reply:
in registerProduct.h

Reply:
Please provide a description of the function. See my previous comment.

Reply:
Description added.  Thanks.